getlatest.go

```diff
@@ -0,0 +1,135 @@
+package namesys
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	ggio "github.com/gogo/protobuf/io"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	pb "github.com/libp2p/go-libp2p-pubsub-router/pb"
+)
+
+type getLatestProtocol struct {
```

Comment:
Can we pick a better name for this type?

Reply:
Any suggestions? I don't have a strong preference. Perhaps tie the name to whatever you're thiniking the protocol name should be (https://github.com/libp2p/go-libp2p-pubsub-router/pull/33#discussion_r312697377).